zebra-chain/src/parameters/network_upgrade.rs

```diff
@@ -118,15 +119,60 @@ const FAKE_TESTNET_ACTIVATION_HEIGHTS: &[(block::Height, NetworkUpgrade)] = &[
 
 /// The Consensus Branch Id, used to bind transactions and blocks to a
 /// particular network upgrade.
-#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    Default,
+    Eq,
+    Hash,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    serde::Serialize,
+    serde::Deserialize,
+)]
+
 pub struct ConsensusBranchId(u32);
 
+impl ConsensusBranchId {
+    /// Return the hash bytes in big-endian byte-order suitable for printing out byte by byte.
+    ///
+    /// Zebra displays transaction and block hashes in big-endian byte-order,
+    /// following the u256 convention set by Bitcoin and zcashd.
+    fn bytes_in_display_order(&self) -> [u8; 4] {
+        let mut reversed_bytes = self.0.to_le_bytes();
+        reversed_bytes.reverse();
+        reversed_bytes
+    }
+}
+
 impl From<ConsensusBranchId> for u32 {
     fn from(branch: ConsensusBranchId) -> u32 {
         branch.0
     }
 }
 
+impl ToHex for &ConsensusBranchId {
+    fn encode_hex<T: FromIterator<char>>(&self) -> T {
+        self.bytes_in_display_order().encode_hex()
+    }
+
+    fn encode_hex_upper<T: FromIterator<char>>(&self) -> T {
+        self.bytes_in_display_order().encode_hex_upper()
+    }
+}
+
+impl FromHex for ConsensusBranchId {
+    type Error = <[u8; 4] as FromHex>::Error;
+
+    fn from_hex<T: AsRef<[u8]>>(hex: T) -> Result<Self, Self::Error> {
+        let branch = <[u8; 4]>::from_hex(hex)?;
+
+        Ok(ConsensusBranchId(u32::from_le_bytes(branch)))
```

Comment:
Question: why are the bytes reversed in ToHex, but not in FromHex?

Reply:
https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/block/hash.rs#L53-L61

https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/block/hash.rs#L73-L81

Reply:
We should probably change above as well. Ill wait for @conradoplg before making changes here.

Reply:
Created a separate PR for it if you need it here: https://github.com/ZcashFoundation/zebra/pull/3893